### DISCUSSION TITLE
How to store and resume?

### DISCUSSION THREAD

**Discussion by jerronl** [aeon-toolkit/aeon#649] [Q&A] [answered]
If is takes very long time to fit a model, can we store the current state to disk and resume later? It would be extremely useful if we run in the google colab.
Thanks!

**Comment by TonyBagnall** [comment, marked as answer]
hi, not currently, but for many of the classifiers at least its not that hard to do. In our java versions we call this checkpointing, where you basically have the option to configure to pickle/serialise the object at regular intervals. We should add this to our road map. What type of estimators are you using?

**Reply by MatthewMiddlehurst** [reply]
This is currently still something we want to do, but have not yet started.

**Comment by jerronl** [comment]
Thanks for your reply. I just started and blindly tried Rocket and miniRocket. Other than the time consumed, I also experienced out of memory crash. Can we also do incremental fit, i.e., each time fit a fraction (batch) of the train data, and gradually improve the gradients?
